extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/InvokerFactoryBuildItem.java

```diff
@@ -0,0 +1,25 @@
+package io.quarkus.arc.deployment;
+
+import org.jboss.jandex.MethodInfo;
+
+import io.quarkus.arc.processor.BeanInfo;
+import io.quarkus.arc.processor.InvokerBuilder;
+import io.quarkus.arc.processor.InvokerFactory;
+import io.quarkus.builder.item.SimpleBuildItem;
+
+/**
+ * Holds an ArC {@link InvokerFactory}. May only be used in the bean registration phase,
+ * observer registration phase, and validation phase; afterwards, any registered invokers
+ * are ignored. TODO this should be enforced somehow?
```

Comment:
Any ideas how to enforce this [using build items probably]? I come up empty :shrug:

Reply:
I don't think you can enforce the usage of a build item, except for throwing an exception... Well, you can store the `BeanDeployment`  reference instead of `InvokerFactory` and throw an exception directly from the `BeanDeployment#getInvokerFactory()`.

I.e. something like:
```java
public InvokerBuilder createInvoker(BeanInfo bean, MethodInfo method) {
        return beanDeployment.getInvokerFactory().createInvoker(bean, method);
}
```

Reply:
Hmm yeah, that makes sense. Thanks!

Reply:
Done.